### DISCUSSION TITLE
What is workdir in overlay volume mounts?

### DISCUSSION THREAD

**Discussion by rEnr3n** [containers/podman#15405] [Q&A] [answered]
```
$ podman --version   
podman version 4.2.0
```

I ran the following commands trying to understand it:

```
$ mkdir dir1
$ mkdir dir2
$ podman run -it --rm -v dir1:/dir1:O,upperdir=$PWD/dir2,workdir=$PWD/dir1 alpine

# pwd
/
# exit

$ podman run -it --rm -v dir1:/dir1:O,upperdir=$PWD/dir2,workdir=/dir1 alpine 
Error: crun: mount `/home/ren/.local/share/containers/storage/overlay-containers/74ce44508f0277aee0df1835f9e2ab215d8e6ad4a7bf17652697c99da444f356/userdata/overlay/3196682081/merge` to `/dir1`: No such file or directory: OCI runtime attempted to invoke a command that was not found

$ podman run -it --rm -v dir1:/dir1:O,upperdir=$PWD/dir2,workdir=$PWD/dir2 alpine
Error: OCI runtime error: crun: mount `/home/ren/.local/share/containers/storage/overlay-containers/dbaae967c563fdf44ea92590ee75b0316a9fa0b5092188dc4b3c17fcf6547357/userdata/overlay/613033634/merge` to `/dir1`: Invalid argument
```

I assumed the workdir here is the same as `--workdir`. Or is it different?

**Comment by mheon** [comment]
@rhatdan @flouthoc PTAL

**Reply by rhatdan** [reply]
--workdir is the default directory inside of the container to place the init process in. IE Current Working Directory
workdir from an Overlay perspective is something different, it is the working directory for the kernel to place content in.  I think this directory has to be empty.  upperdir will contain content that is written to the upperlevel of the overlay.  workdir content is not defined for overlay.

**Comment by flouthoc** [comment, marked as answer]
@rEnr3n , This is correct https://github.com/containers/podman/discussions/15405#discussioncomment-3456607 , `--workdir` is different from overlay's advanced option `workdir=`. Overlay's workdir options is used to prepare files before they are switched to the overlay destination in an atomic action while `podman`'s `--workdir` is completely different option as @rhatdan stated in above comment.

Both options are completely different and not directly related to each other.
